flight/Libraries/mgrs/polarst.c

```diff
@@ -0,0 +1,523 @@
+ *    coordinates.
+ *
+ * ERROR HANDLING
+ *
+ *    This component checks parameters for valid values.  If an invalid 
+ *    value is found the error code is combined with the current error code 
+ *    using the bitwise or.  This combining allows multiple error codes to 
+ *    be returned. The possible error codes are:
+ *
+ *          POLAR_NO_ERROR           : No errors occurred in function
+ *          POLAR_LAT_ERROR          : Latitude outside of valid range
+ *                                      (-90 to 90 degrees)
+ *          POLAR_LON_ERROR          : longitude outside of valid range
+ *                                      (-180 to 360 degrees) 
+ *          POLAR_ORIGIN_LAT_ERROR   : Latitude of true scale outside of valid
+ *                                      range (-90 to 90 degrees)
+ *          POLAR_ORIGIN_LON_ERROR   : longitude down from pole outside of valid
+ *                                      range (-180 to 360 degrees)
+ *          POLAR_EASTING_ERROR      : Easting outside of valid range,
+ *                                      depending on ellipsoid and
+ *                                      projection parameters
+ *          POLAR_NORTHING_ERROR     : Northing outside of valid range,
+ *                                      depending on ellipsoid and
+ *                                      projection parameters
+ *          POLAR_RADIUS_ERROR       : Coordinates too far from pole,
+ *                                      depending on ellipsoid and
+ *                                      projection parameters
+ *          POLAR_A_ERROR            : Semi-major axis less than or equal to zero
+ *          POLAR_INV_F_ERROR        : Inverse flattening outside of valid range
+ *								  	                  (250 to 350)
+ *
+ *
+ * REUSE NOTES
+ *
+ *    POLAR STEREOGRAPHIC is intended for reuse by any application that  
+ *    performs a Polar Stereographic projection.
+ *
+ *
+ * REFERENCES
+ *
+ *    Further information on POLAR STEREOGRAPHIC can be found in the
+ *    Reuse Manual.
+ *
+ *
+ *    POLAR STEREOGRAPHIC originated from :
+ *                                U.S. Army Topographic Engineering Center
+ *                                Geospatial Information Division
+ *                                7701 Telegraph Road
+ *                                Alexandria, VA  22310-3864
+ *
+ *
+ * LICENSES
+ *
+ *    None apply to this component.
+ *
+ *
+ * RESTRICTIONS
+ *
+ *    POLAR STEREOGRAPHIC has no restrictions.
+ *
+ *
+ * ENVIRONMENT
+ *
+ *    POLAR STEREOGRAPHIC was tested and certified in the following
+ *    environments:
+ *
+ *    1. Solaris 2.5 with GCC, version 2.8.1
+ *    2. Window 95 with MS Visual C++, version 6
+ *
+ *
+ * MODIFICATIONS
+ *
+ *    Date              Description
+ *    ----              -----------
+ *    06-11-95          Original Code
+ *    03-01-97          Original Code
+ *
+ *
+ */
```

Comment:
Same question as above. It's not problem to merge in (I found the "LICENSES:  None apply to this component."), but for posterity it would be good to know who was the original author.